docker-images/kafka-based/kafka/scripts/kafka_run.sh

```diff
@@ -93,13 +93,13 @@ if [ "$USE_KRAFT" == "true" ]; then
   KAFKA_READY=
   ZK_CONNECTED=
 else
-  KRAFT_LOG_DIR=$(grep "log\.dirs=" /tmp/strimzi.properties | sed "s/log\.dirs=*//")
-
   # when in ZooKeeper mode, the __cluster_metadata folder should not exist.
   # if it does, it means a KRaft migration rollback is ongoing and it has to be removed.
-  if [ -d "$KRAFT_LOG_DIR/__cluster_metadata-0" ]; then
+  # also checking that metadata state is ZK (0), because if it's MIGRATION (2) it means we are rolling back but not finalized yet and KRaft quorum is still in place.
+  CURRENT_KRAFT_METADATA_LOG_DIR=$(ls -d /var/lib/kafka/data*/kafka-log"$STRIMZI_BROKER_ID"/__cluster_metadata-0 2> /dev/null || true)
```

Comment:
using `data*` and not `data-*` because of the folder name when we don't use JBOD but just `persistent-claim` storage type.

Reply:
Are you aware of any actual bug? Or are you just increasing the testing scope here? The logic is that unless you use JBOD, there is no moving of the metadata directories. So we do not care about the non-JBOD storage.